java-runtime/src/test/kotlin/io/spine/validate/EnclosedMessageValidationSpec.kt

```diff
@@ -0,0 +1,149 @@
+import io.spine.test.validate.PatternStringFieldValue
+import io.spine.test.validate.notValidateEnclosed
+import io.spine.validate.ValidationOfConstraintTest.Companion.VALIDATION_SHOULD
+import io.spine.validate.given.MessageValidatorTestEnv.EMAIL
+import io.spine.validate.given.MessageValidatorTestEnv.ENCLOSED_FIELD_NAME
+import io.spine.validate.given.MessageValidatorTestEnv.assertFieldPathIs
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+
+@DisplayName(VALIDATION_SHOULD + "validate enclosed messages and")
+internal class EnclosedMessageValidationSpec : ValidationOfConstraintTest() {
+
+    @Test
+    fun `find out that enclosed message field is valid`() {
+        val enclosedMsg = PatternStringFieldValue.newBuilder()
+            .setEmail("valid.email@mail.com")
+            .build()
+        val msg = ValidateEnclosed.newBuilder()
+            .setEnclosed(enclosedMsg)
+            .build()
+        assertValid(msg)
+    }
+
+    @Test
+    fun `find out enclosed message field is NOT valid`() {
+        val enclosedMsg = PatternStringFieldValue.newBuilder()
+            .setEmail("invalid email")
+            .buildPartial()
+        val msg = ValidateEnclosed.newBuilder()
+            .setEnclosed(enclosedMsg)
+            .buildPartial()
+        assertNotValid(msg)
+    }
+
+    @Test
+    fun `consider field valid if no valid option is set`() {
+        val enclosedMsg: @NonValidated PatternStringFieldValue =
+            PatternStringFieldValue.newBuilder()
+                .setEmail("invalid email")
+                .buildPartial()
+        val msg = notValidateEnclosed {
+            enclosed = enclosedMsg
+        }
+        assertValid(msg)
+    }
+
+    @Test
+    fun `consider field valid if it is not set`() {
+        val msg = ValidateWithRequiredString.getDefaultInstance()
+        assertValid(msg)
+    }
+
+    @Test
+    @Disabled("Until 'skipValidation()` is turned off.")
+    fun `provide valid violations if enclosed message field is not valid`() {
+        val enclosedMsg = PatternStringFieldValue.newBuilder()
+            .setEmail("invalid email")
+            .buildPartial()
+        val msg = ValidateEnclosed.newBuilder()
+            .setEnclosed(enclosedMsg)
+            .buildPartial()
+        validate(msg)
+
+        val violation = singleViolation()
+
+        violation.msgFormat shouldBe "The message must have valid properties."
+
+        assertFieldPathIs(
+            violation,
+            ENCLOSED_FIELD_NAME
+        )
+        val innerViolations = violation.violationList
+        innerViolations shouldHaveSize  1
+
+        val innerViolation = innerViolations[0]
+
+        innerViolation.msgFormat shouldStartWith Diags.Regex.prefix
+
+        assertFieldPathIs(
+            innerViolation,
+            EMAIL
+        )
+
+        innerViolation.violationList.shouldBeEmpty()
```

Comment:
There are many empty lines, and it is quite hard to understand grouping. Maybe reduce number of empty lines or extract smaller asserting methods?